src/nupic/types/Serializable.hpp

```diff
@@ -273,11 +287,17 @@ class Serializable {
   }
   
   virtual inline void loadFromFile_ar(std::string filePath, SerializableFormat fmt=SerializableFormat::BINARY) {
-    std::ifstream in(filePath, std::ios_base::in | std::ios_base::binary);
-    in.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+		std::ios_base::openmode mode = std::ios_base::in;
+		if (fmt <= SerializableFormat::PORTABLE) mode |= std::ios_base::binary;
+	  std::ifstream in(filePath, mode);
+    // NOTE: do NOT set stream exceptions:
+    //       in.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+    //       The JSON parser will not be able to find the end of the parse.
 		loadFromStream_ar(in, fmt);
 		in.close();
 	}
+  // NOTE: for BINARY and PORTABLE the stream must opened with ios_base::binary or it will crash on Windows.
```

Comment:
OT: ditto, file a bug?

Reply:
> BINARY and PORTABLE the stream must opened with ios_base::binary or it will crash on Windows
OT: ditto, file a bug?

Also not a bug.  Binary implies it needs a binary stream.